src/totp/totp.cpp

```diff
@@ -93,30 +91,53 @@ QString QTotp::parseOtpString(QString key, quint8 &digits, quint8 &step)
     return seed;
 }
 
-QString QTotp::generateTotp(const QByteArray key, quint64 time,
-                            const quint8 numDigits = defaultDigits, const quint8 step = defaultStep)
+QString QTotp::generateTotp(const QByteArray key,
+                            quint64 time,
+                            const quint8 numDigits = defaultDigits,
+                            const quint8 step = defaultStep)
 {
     quint64 current = qToBigEndian(time / step);
 
-    QByteArray secret = Base32::base32_decode(key);
-    if (secret.isEmpty()) {
+    Optional<QByteArray> secret = Base32::decode(Base32::sanitizeInput(key));
+    if (!secret.hasValue()) {
         return "Invalid TOTP secret key";
     }
 
     QMessageAuthenticationCode code(QCryptographicHash::Sha1);
-    code.setKey(secret);
+    code.setKey(secret.valueOr(""));
     code.addData(QByteArray(reinterpret_cast<char*>(&current), sizeof(current)));
     QByteArray hmac = code.result();
 
     int offset = (hmac[hmac.length() - 1] & 0xf);
-    int binary =
-            ((hmac[offset] & 0x7f) << 24)
-            | ((hmac[offset + 1] & 0xff) << 16)
-            | ((hmac[offset + 2] & 0xff) << 8)
-            | (hmac[offset + 3] & 0xff);
+    int binary = ((hmac[offset] & 0x7f) << 24) | ((hmac[offset + 1] & 0xff) << 16) | ((hmac[offset + 2] & 0xff) << 8) |
```

Comment:
I liked the previous formatting better. It's hard to read on a single line.

Reply:
I agree with you, this was the result of running clang-format; I'll change it.